### ISSUE TITLE
Enhanced stats output from LUA hook

### ISSUE THREAD

**Issue by Aranud** [etlegacy/etlegacy#2974]
Add severals hook related to statistic to improved usage outside game.

Linked to https://github.com/etlegacy/etlegacy/issues/2703 as we can improved output from this point too.

Find all the occurence of Log usage to trace stats gathering and use the appropriated hook.

**Comment by csmaxpower** [comment]
For reference, the original request for JSON output in #2703 can be accomplished in LUA now and have synced up with @utrolig on it.  It would still be nice to have additional hooks as this (#2974) issue mentions for more detailed stats for things that are seen in console events that aren't tracked otherwise in `g_match.c` or even some native JSON output for modern day tracking.  

The Lua engine has the ability to use `curl` so you can simply `POST` a built payload to some API endpoint as seen below.  The only caveat is that depending on the complexity or level of JSON manipulation you want to do in the Lua itself, could require additional libraries that don't come with the servers and engine by default.

Also for note, using `curl` and `POST` in Lua without some thought around async and endpoint responses, can cause the server to have hiccups in frame processing while waiting on those responses script side, so, be careful XD.  

**Example Lua for Posting JSON**
```Lua

-- Combine the header and stats
local final_data = {
    round_info = header_json,
    player_stats = stats_json
}
-- JSON encode the table (you may need to include a JSON library such as dkjson)
local json = require ("dkjson")
local stats_json = json.encode(final_data)

-- Escape single quotes in JSON data
stats_json = escape_single_quotes(stats_json)

-- post the JSON data to an API using curl
local api_url = "https://api.moestavern.pub/stats"
local handle = io.popen('curl -X POST -H "Content-Type: application/json" -d \'' .. stats_json .. '\' ' .. api_url)
```

**Example JSON Stats Output**
```JSON
{
    "header":{
       "timelimit":"10:00",
       "nextTimeLimit":"0:50",
       "servername":"^eMoe^7'^es ^7Legacy ^dScrims ^7NA ^iATL",
       "config":"practice",
       "winnerteam":2,
       "mapname":"adlernest",
       "matchID":"1247",
       "round":2,
       "defenderteam":2
       },
    "stats":{
       "123456789101112131415161718":{
          "guid":"1234567",
          "selfkills":0,
          "team":1,
          "damageReceived":0,
          "timePlayed":100.0,
          "weaponStats":[
             {
                "hits":4,
                "atts":6,
                "kills":2,
                "deaths":0,
                "headshots":0
             },
             {
                "hits":1,
                "atts":8,
                "kills":1,
                "deaths":0,
                "headshots":0
             },
             {
                "hits":0,
                "atts":19,
                "kills":0,
                "deaths":0,
                "headshots":0
             },
             {
                "hits":0,
                "atts":4,
                "kills":1,
                "deaths":0,
                "headshots":0
             }
          ],
          "gibs":0,
          "teamgibs":0,
          "name":"^d[^0moes^d]^0mAxPower^d.",
          "dwWeaponMask":2090,
          "rounds":2,
          "damageGiven":0,
          "teamDamageReceived":0,
          "teamkills":0,
          "xp":24,
          "teamDamageGiven":0
       }
    }
 }
 ```
